sdk/core/Azure.Core/src/Http/ConditionalRequestOptions.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Http
```

Comment:
I'm kind of on the fence about whether we want these here or `Azure`.  For your scenarios, `Azure.Core.Http` makes more sense since this is the advanced option.  I think they'll be used more often in Storage.

I could live with either, but I'd invite others to weigh in if they have a strong preference.

Reply:
Depending on what we decide the rules are. If we use rule 1 from proposed https://gist.github.com/pakrym/6371cd8bce13f64d2e00dd2110a48058 it has to go into Azure.

Reply:
Tracking in issue #7895